agent/xds/testdata/endpoints/mesh-gateway-service-subsets.latest.golden

```diff
@@ -143,124 +143,14 @@
       "@type": "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment",
       "clusterName": "v1.bar.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul",
       "endpoints": [
-        {
```

Comment:
This change came because there were two tests named the same:
- mesh-gateway-service-subsets in [endpoints_test.go](https://github.com/hashicorp/consul/pull/19502/files#diff-2124eb0e3f144aacf76043caae464f2f29c94303c4b50a1ef454b6312b02d730L415) that was running `service-subsets2` for endpoints only
- mesh-gateway-service-subsets in [clusters_test.go](https://github.com/hashicorp/consul/pull/19502/files#diff-7bcb66e6b8a79cecf0a24c58e45f62fdd3974f309ccdd359f62b796a621a42ecL609) that is running `service-subsets`

The test case running `service-subsets2` has been added as `mesh-gateway-service-subsets2` and this diff is because this file now shows the `mesh-gateway-service-subsets` test that was taken from clusters_test.go and the deleted output is now in the `mesh-gateway-service-subsets2` test case file.